### ISSUE TITLE
unexpected behaviour of dashed bibliography entries

### ISSUE THREAD

**Issue by wujastyk** [alex-ball/biblatex-oxref#9]
Using TeXlive 2019 updated to the present; oxyear.dbx 2019/09/02 v1.2; xelatex.

I have 
```
style=oxyear,
dashed=true,
```
But in the final bibliography, after many reruns of latex, I still have multiple author entries where I would expect dashes.  I can't see a pattern yet, unless it's that the name gets printed every time the type of publication switches (incollection, article, etc.)?

![Screenshot from 2019-09-12 00-50-06](https://user-images.githubusercontent.com/762246/64760655-7a935f80-d4f7-11e9-8306-cfa269b4dfe3.png)

**Comment by alex-ball** [comment]
Please could you supply me with a .bib file containing the affected entries? This will make it a lot easier for me to (a) see what is going on, and (b) test any fixes I might introduce.

**Comment by wujastyk** [comment]
Attached.  Many thanks for having a look into this!

**Comment by alex-ball** [comment]
I think GitHub must have stripped out the attachment from your email. Could you please see if you can attach the file via the web interface at <https://github.com/alex-ball/biblatex-oxref/issues/9>?

**Comment by wujastyk** [comment]
Here it is, gzipped (.bib isn't an accepted filetype for github attachments)

[Detailed-description.bib.gz](https://github.com/alex-ball/biblatex-oxref/files/3610678/Detailed-description.bib.gz)

**Comment by alex-ball** [comment]
I have identified the cause. I will need to do some testing before putting this out in a new release, but in the meantime, here's a fix.

After loading biblatex with the oxyear style, add these lines:

```tex
\makeatletter
\xpatchbibmacro{bookeditor}{\global\undef\bbx@lasthash}{}{}{%
  \wlog{INFO: oxyear fix no longer needed!}}
\makeatother
```

Assuming my testing goes as expected, when the fix is released, the `\wlog` command will write the INFO line in your log so you know you can remove the fix again.

**Comment by wujastyk** [comment]
That's great, and it works.

I had some trouble at first because I put in the makeatletter/-other
commands without realizing that I was writing in a .sty file (where I set
up biblatex stuff).  So that \makeatother command turned off @ for the rest
of the sty file, generating many cryptic errors.  Anyhow, I sorted it out
eventually.

I didn't know about xpatch before; very powerful and sophisticated!

Thanks again,
Dominik
